src/client/actions/gaEvents/index.tsx

```diff
@@ -0,0 +1,37 @@
+import ReactGA from 'react-ga'
+import { get } from '../../util/requests'
+
+// Checks if reactga has initalised
+let hasInit = false
+
+const getGaId = async () => {
+    await get('/api/ga').then(async (response) => {
+      if(response.ok) {
+        await response.text().then((gaid) => {
+            ReactGA.initialize(gaid)
+            hasInit = true
+        })
+      }
+    })
+  }
+// Records clicks - customise action and label name to record multiple uniques per session
+export const GAevent = async (categoryName:string, eventName:string, label?: string| undefined) => {
+    if (!hasInit) {
```

Comment:
you could cheat and inspect the presence of `window.ga` which is what React does. This however would be a bit of a code smell, as you now need knowledge of the presence of that global.